application/src/main/java/org/togetherjava/tjbot/commands/utils/StringDistances.java

```diff
@@ -73,14 +81,14 @@ public static Collection<String> closeMatches(CharSequence prefix,
         bestMatches.addAll(scoredMatches);
 
         return Stream.generate(bestMatches::poll)
+            .limit(limit)
             .takeWhile(matchScore -> isCloseEnough(matchScore, prefix))
             .map(MatchScore::candidate)
-            .limit(limit)
             .toList();
     }
 
     private static boolean isCloseEnough(MatchScore matchScore, CharSequence prefix) {
-        return matchScore.score / prefix.length() <= OFF_BY_PERCENTAGE_THRESHOLD;
+        return (double) matchScore.score / prefix.length() <= OFF_BY_PERCENTAGE_THRESHOLD;
```

Comment:
voting for `score` itself to be `double` aready. based on the name, one would except that at least.

Reply:
when PED method returns `int` it should be `int` ig but idk, i'll make it a `double` if u want

Reply:
the PED should stay int but the score itself should be double. in the future, it might also include relevance or other stuff, BM-25 matching etc blabla. it doesnt necessarily have to be a PED only. otherwise, the name should be changed from score to distance imo - then it could stay an int 🤷